### PR TITLE
Host Command improvements

### PR DESCRIPTION
**Pull request by niedzwiecki-dawid**
A few improvements:
-make the Host Command structure available,
-follow clang format
-add autoinit config
-update checking buffers pointers

### CODE REVIEW

File: include/zephyr/mgmt/ec_host_cmd/ec_host_cmd.h [hunk 1]
```diff
@@ -231,8 +232,8 @@ enum ec_host_cmd_status {
  * This routine initializes the host command subsystem. It includes initialization
  * of a backend and the handler.
  * When the application configures the zephyr,host-cmd-espi-backend/zephyr,host-cmd-shi-backend/
- * zephyr,host-cmd-uart-backend chosen node, the chosen backend automatically
- * calls this routine at CONFIG_EC_HOST_CMD_INIT_PRIORITY.
+ * zephyr,host-cmd-uart-backend chosen node and CONFIG_EC_HOST_CMD_INITIALIZE_AT_BOOT is set,
```
Comment (nordicjm) [inline include/zephyr/mgmt/ec_host_cmd/ec_host_cmd.h:235]:
```suggestion
 * zephyr,host-cmd-uart-backend chosen node and @kconfig{CONFIG_EC_HOST_CMD_INITIALIZE_AT_BOOT} is set,
```

File: include/zephyr/mgmt/ec_host_cmd/ec_host_cmd.h [hunk 2]
```diff
@@ -231,8 +232,8 @@ enum ec_host_cmd_status {
...
+ * the chosen backend automatically calls this routine at CONFIG_EC_HOST_CMD_INIT_PRIORITY.
```
Comment (nordicjm) [inline include/zephyr/mgmt/ec_host_cmd/ec_host_cmd.h:236]: As above

File: subsys/mgmt/ec_host_cmd/Kconfig [hunk 3]
```diff
@@ -74,6 +74,13 @@ config EC_HOST_CMD_HANDLER_RX_BUFFER_DEF
 	help
 	  The handler defines common rx buffer
 
+config EC_HOST_CMD_INITIALIZE_AT_BOOT
+	bool "Initialize the host command subsystem automacitly"
```
Comment (nordicjm) [inline subsys/mgmt/ec_host_cmd/Kconfig:78]: *automatically

Comment (nordicjm) [inline subsys/mgmt/ec_host_cmd/Kconfig:78]: *automatically

**Review by nordicjm — CHANGES_REQUESTED** [review]
Only doc text checked

File: include/zephyr/mgmt/ec_host_cmd/ec_host_cmd.h (hunk 1 above)
Comment (semihalf-niedzwiecki-dawid) [reply include/zephyr/mgmt/ec_host_cmd/ec_host_cmd.h]: Done.

File: subsys/mgmt/ec_host_cmd/Kconfig (hunk 3 above)
Comment (semihalf-niedzwiecki-dawid) [reply subsys/mgmt/ec_host_cmd/Kconfig]: Done.

File: subsys/mgmt/ec_host_cmd/ec_host_cmd_handler.c [hunk 4]
```diff
@@ -244,20 +245,28 @@ int ec_host_cmd_init(struct ec_host_cmd_backend *backend)
...
-	if ((handler_tx_buf && (handler_tx_buf != hc->tx.buf)) ||
-	    (handler_rx_buf && (handler_rx_buf != hc->rx_ctx.buf))) {
+	/* Check if a backend uses provided buffers. The buffer pointers can be shifted within the
+	 * buffer to make space for preamble. Make sure the rx/tx pointers are within the provided
+	 * buffers ranges.
+	 */
+	if ((handler_tx_buf && !((handler_tx_buf <= (uint8_t *)hc->tx.buf) &&
+				(tx_buf_end > (uint8_t *)hc->tx.buf))) ||
+	    (handler_rx_buf && !((handler_rx_buf <= hc->rx_ctx.buf) &&
+				(rx_buf_end < hc->rx_ctx.buf)))) {
```
Comment (fabiobaltieri) [inline subsys/mgmt/ec_host_cmd/ec_host_cmd_handler.c:269]: Would it make sense to have another pair of local variables for `hc->tx.buf` and `hc->rx_ctx.buf`?  Something descriptive, just trying to make the condition easier to understand, it's a bit puzzling that first you set them to `handler_tx_buf` but then they change from `init()`... maybe `backend_tx_buf` and `backend_rx_buf`?

Comment (semihalf-niedzwiecki-dawid) [reply subsys/mgmt/ec_host_cmd/ec_host_cmd_handler.c]: Make sense if you think it is easier to understand.

**Review by fabiobaltieri — DISMISSED** [review]

File: doc/services/device_mgmt/ec_host_cmd.rst [hunk 5]
```diff
@@ -47,8 +47,9 @@ The supported backend and peripheral drivers:
 Initialization
 **************
 
-If the application configures one of the following backend chosen nodes, then the corresponding backend
-initializes the host command subsystem by calling :c:func:`ec_host_cmd_init`:
+If the application configures one of the following backend chosen nodes and
+:kconfig:option:`CONFIG_EC_HOST_CMD_INITIALIZE_AT_BOOT` is set, then the corresponding backend initializes the host
```
Comment (keith-zephyr) [inline doc/services/device_mgmt/ec_host_cmd.rst:51]: Nit - line length should be less than 100.

File: subsys/mgmt/ec_host_cmd/ec_host_cmd_handler.c [hunk 6]
```diff
@@ -244,20 +246,31 @@ int ec_host_cmd_init(struct ec_host_cmd_backend *backend)
...
 
+	backend_tx_buf = hc->tx.buf;
+	backend_rx_buf = hc->rx_ctx.buf;
+
 	if (ret != 0) {
 		return ret;
 	}
 
-	if (!hc->tx.buf | !hc->rx_ctx.buf) {
+	if (!hc->tx.buf || !hc->rx_ctx.buf) {
```
Comment (keith-zephyr) [inline subsys/mgmt/ec_host_cmd/ec_host_cmd_handler.c:261]: Use your helper variable consistently.
```suggestion
	if (!backend_tx_buf|| !backend_rx_buf) {
```

**Review by keith-zephyr — CHANGES_REQUESTED** [review]

File: subsys/mgmt/ec_host_cmd/ec_host_cmd_handler.c (hunk 6 above)
Comment (semihalf-niedzwiecki-dawid) [reply subsys/mgmt/ec_host_cmd/ec_host_cmd_handler.c]: Done.

File: doc/services/device_mgmt/ec_host_cmd.rst (hunk 5 above)
Comment (semihalf-niedzwiecki-dawid) [reply doc/services/device_mgmt/ec_host_cmd.rst]: Done.

**Review by fabiobaltieri — APPROVED** [review]

**Review by keith-zephyr — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Pull request merged**